src/Interpreters/Set.cpp

```diff
@@ -278,6 +280,21 @@ void Set::checkIsCreated() const
         throw Exception(ErrorCodes::LOGICAL_ERROR, "Trying to use set before it has been built.");
 }
 
+ColumnPtr returnFilteredColumn(const ColumnPtr & first, const ColumnPtr & second)
+{
+    ConstantFilterDescription second_const_descr(*second);
+    if (second_const_descr.always_true)
+        return second;
+
+    if (second_const_descr.always_false)
+        return first;
+
+    FilterDescription filter_descr(*second);
+    if (filter_descr.data)
```

Comment:
I think `.data` should never be nullptr. Probably we can change it in FilterDescription to ref.